src/components/Dropdown/DropdownList.tsx

```diff
@@ -56,21 +57,98 @@ export const DropdownList = (props: IDropdownListProps): ReactElement => {
     ],
   });
 
+  const close = () => {
+    if (visible) {
+      setVisible(false);
+      onClosed();
+    }
+  };
+
+  const open = () => {
+    if (!visible) {
+      setVisible(true);
+      onExpanded();
+    }
+  };
+
+  /* Click on dropdown */
   const handleClick = () => {
     setVisible(!visible);
+    visible ? onExpanded() : onClosed();
   };
 
-  const handleSelect = useCallback(
-    (item: ItemType) => {
-      return () => {
-        onSelect(item.id);
-      };
-    },
-    [onSelect],
-  );
+  /* keydown on dropdown */
+  const handleKeyDown = (e: KeyboardEvent) => {
+    switch (e.key) {
+      case 'Enter':
+        e.preventDefault();
+        handleClick();
+        return;
+      case 'Space':
+        e.preventDefault();
+        handleClick();
+        return;
+      case 'Escape':
+        close();
+        return;
+      case 'ArrowDown':
+        e.preventDefault();
+        open();
+        focusItem(0);
+        return;
+      default:
+        return;
+    }
+  };
+
+  /* selected item */
+  const handleSelect = (item: ItemType) => {
+    onSelect(item.id);
+    close();    
+  };
+
+  /* keydown on item */
+  const handleItemKeyDown = (e: KeyboardEvent, item: ItemType, index: number) => {
```

Comment:
This is okay, I just did a couple edits to make it more concise:
```js
const handleItemKeyDown = (e: KeyboardEvent, item: ItemType, index: number) => {
  switch (e.key) {
    case 'Enter':
    case 'Space':
      e.preventDefault();
      return handleSelect(item);
    case 'ArrowUp':
      e.preventDefault();
      return focusItem(index - 1);
    case 'ArrowDown':
      e.preventDefault();
      return focusItem(index + 1);
    case 'Escape':
    case 'Tab':
      return close();
  }
};
```
With a switch statement we can flow down to subsequent ones if the action is the same, so thats what I did with the 'Enter'.  Also, moved the return to the line above just to clean up a little.

Reply:
Yeah we could keep it, but since you aren't really returning a value other than `undefined` I think we can leave it off and return after the switch.

Reply:
Updated